### PR TITLE
feat: add devcontainer support

### PR DESCRIPTION
**Pull request by ianchen-tw**
## Overview
1. add devcontainer support

### Commands tested

+ `cargo test`
+ `cargo fmt`
+ `cargo clippy`
+ `cargo check`
+ Build with rust
  + `cmake -B build -DCMAKE_BUILD_TYPE=Release -DWITH_RUST=ON`
  + `cmake --build ./build -v --config Release`
  + `cd build && ctest -C Release --output-on-failure`
+ Build without rust
  + `cmake -B build -DCMAKE_BUILD_TYPE=Release`
  + `cmake --build ./build -v --config Release`
  + `cd build && ctest -C Release --output-on-failure`

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
+ARG RUST_VERSION=1.68.2
+ARG CMAKE_VERSION=3.26.3
```
Comment (kanru) [inline .devcontainer/Dockerfile:2]: Do we need to specify the version here?

Rustup will automatically download the version specified in rust-toolchain.toml so the version here is just used for bootstrap.

For cmake, the minimum supported version is 3.19.0. We can probably just use the package provided by the base distro image (debian-11 in this case).

**Comment by ianchen-tw** [comment]
這個PR有幾個已經知道的問題，比如說 rustfmt 這些工具沒有裝好，需要再研究一下

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (ianchen-tw) [reply .devcontainer/Dockerfile]: 722f18b
+ Removed CMake version, using Debian default
+ Synced Rust version with MSVC version to save users time. Users will no longer need to redownload Rust after opening the devcontainer.

**Comment by ianchen-tw** [comment]
@kanru 
我把 msvc 這件事情考慮進去了，如果可以的話再幫我使用 devcontainer 連進去，看看能不能滿足平常的開發流程需求

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"customizations": {
+		"vscode": {
+			"extensions": [
+				"eamodio.gitlens", // IDE Git information
+				"shardulm94.trailing-spaces", // Show trailing spaces
+				"christian-kohler.path-intellisense",
+				"ms-azuretools.vscode-docker",
+				"ms-vscode.cmake-tools", // Cmake
+				// Rust language support
+				"rust-lang.rust", // Use "rust-lang.rust-analyzer" once MSRV reaches 1.64
```
Comment (kanru) [inline .devcontainer/devcontainer.json:18]: If we use `rust-lang.rust-analyzer` should it be able to download the corresponding pre-built version automatically?

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+				// Ref:
+				// + https://github.com/chewing/libchewing/pull/347#issuecomment-1412022376
+				// + https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1640-2022-09-22
+				"tamasfe.even-better-toml", // for Cargo.toml
+				// Other
+				"genieai.chatgpt-vscode"
```
Comment (kanru) [inline .devcontainer/devcontainer.json:24]: I feel this should not be in the default extensions set. Users can install additional extensions if they want.

**Review by kanru — COMMENTED** [review]
Over all LGtM.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (ianchen-tw) [reply .devcontainer/devcontainer.json]: 25208bfe

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (ianchen-tw) [reply .devcontainer/devcontainer.json]: 25208bfe

**Review by kanru — APPROVED** [review]

**Pull request merged**